pkg/process/runner/submitter.go

```diff
@@ -166,7 +167,7 @@ func NewSubmitter(hostname string, enableRealtimeCallback func([]*model.Collecto
 
 		forwarderRetryMaxQueueBytes: queueBytes,
 
-		updateRTStatusCallback: enableRealtimeCallback,
+		rtNotifierChan: make(chan types.RTResponse, 1), // Buffer the channel so we don't block submissions
```

Comment:
Is it worth having a larger channel buffer size, or do we think that this channel should never fill up?

Reply:
I think this channel really shouldn't fill up. The RT check runs every 2 seconds, so at most we only enqueue data onto this channel every 2s. Additionally, the channel is polled pretty frequently, as the runner has its own goroutine to listen for changes ([here](https://github.com/DataDog/datadog-agent/blob/a1b5dc2c7f4f9a7ca30a84d1c2f35888f3ab6590/pkg/process/runner/runner.go#L294-L310)). 

The only scenario I can think of that this would be an issue is if the runner were to be stopped or if RT mode is explicitly disallowed. In this case, nothing is reading on the channel so IMO it's fine to drop those updates.

Let me know if you agree/disagree, I also don't see any issue with bumping the channel to a size of 2, so if you disagree with me I'm happy to increase the size as well.

Reply:
Chatted offline. It seems like this shouldn't be an issue.